Build failure: stuntrally

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

```
nix run nixpkgs#stuntrally
```

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: builder for '/nix/store/cbgw5p1n91gbzhabb0v95va6827hjkjk-stuntrally-2.7.drv' failed with exit code 2;
       last 25 log lines:
       >                  from /build/source/source/shiny/Main/Preprocessor.hpp:16,
       >                  from /build/source/source/shiny/Main/Preprocessor.cpp:2:
       > include/boost/detail/atomic_count.hpp:10:1: note: '#pragma message: This header is deprecated. Use <boost/smart_ptr/detail/atomic_count.hpp> instead.'
       > [ 14%] Building CXX object source/paged-geom/CMakeFiles/paged-geom.dir/WindBatchedGeometry.cpp.o
       > [ 14%] Building CXX object source/shiny/CMakeFiles/shiny.dir/Main/ShaderInstance.cpp.o
       > [ 14%] Building CXX object source/shiny/CMakeFiles/shiny.dir/Main/ShaderSet.cpp.o
       > [ 14%] Linking CXX static library libpaged-geom.a
       > [ 14%] Built target paged-geom
       > In file included from include/boost/smart_ptr/detail/deprecated_macros.hpp:8,
       >                  from include/boost/smart_ptr/detail/sp_counted_base.hpp:22,
       >                  from include/boost/smart_ptr/detail/shared_count.hpp:22,
       >                  from include/boost/smart_ptr/shared_ptr.hpp:17,
       >                  from include/boost/shared_ptr.hpp:17,
       >                  from /build/source/source/shiny/Main/Platform.hpp:6,
       >                  from /build/source/source/shiny/Main/ShaderInstance.hpp:6,
       >                  from /build/source/source/shiny/Main/ShaderInstance.cpp:2:
       > include/boost/bind.hpp:36:1: note: '#pragma message: The practice of declaring the Bind placeholders (_1, _2, ...) in the global namespace is deprecated. Please use <boost/bind/bind.hpp> + using namespace boost::placeholders, or define BOOST_BIND_GLOBAL_PLACEHOLDERS to retain the current behavior.'
       > /build/source/source/shiny/Main/ShaderSet.cpp: In constructor 'sh::ShaderSet::ShaderSet(const std::string&, const std::string&, const std::string&, const std::string&, const std::string&, const std::string&, sh::PropertySetGet*)':
       > /build/source/source/shiny/Main/ShaderSet.cpp:32:23: error: 'class boost::filesystem::path' has no member named 'branch_path'
       >    32 |                 p = p.branch_path();
       >       |                       ^~~~~~~~~~~
       > make[2]: *** [source/shiny/CMakeFiles/shiny.dir/build.make:205: source/shiny/CMakeFiles/shiny.dir/Main/ShaderSet.cpp.o] Error 1
       > make[2]: *** Waiting for unfinished jobs....
       > make[1]: *** [CMakeFiles/Makefile2:519: source/shiny/CMakeFiles/shiny.dir/all] Error 2
       > make: *** [Makefile:136: all] Error 2
       For full logs, run 'nix log /nix/store/cbgw5p1n91gbzhabb0v95va6827hjkjk-stuntrally-2.7.drv'.
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.14.0, NixOS, 25.05 (Warbler), 25.05.20250331.77b584d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.13`
 - nixpkgs: `/nix/store/w7ncjpwxd9hj1bjqi3wj0h1mp2d5pj86-source`

### Notify maintainers


@pSub
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -17740,7 +17740,7 @@ with pkgs;
 
   protonup-ng = with python3Packages; toPythonApplication protonup-ng;
 
-  stuntrally = callPackage ../games/stuntrally { };
+  stuntrally = callPackage ../games/stuntrally { boost = boost183; };
 
   superTuxKart = darwin.apple_sdk_11_0.callPackage ../games/super-tux-kart {
     inherit (darwin.apple_sdk_11_0.frameworks)

